ddtrace/llmobs/_llmobs.py

```diff
@@ -573,6 +575,42 @@ def create_dataset(cls, name: str, description: str) -> Dataset:
     def _delete_dataset(cls, dataset_id: str) -> None:
         return cls._instance._dne_client.dataset_delete(dataset_id)
 
+    @classmethod
+    def experiment(
+        cls,
+        name: str,
+        task: Callable[[Dict[str, Any]], Any],
+        dataset: Dataset,
+        evaluators: List[Callable[[Any, Any, Any], Any]],
+        description: str = "",
+    ) -> Experiment:
+        """Initializes an Experiment to run a task on a Dataset and evaluators.
+
+        :param name: The name of the experiment.
+        :param task: The task function to run. Must accept a parameter ``input_data`` and optionally ``config``.
+        :param dataset: The dataset to run the experiment on, created with LLMObs.pull/create_dataset().
+        :param evaluators: A list of evaluator functions to evaluate the task output.
+                           Must accept parameters ``input_data``, ``output_data``, and ``expected_output``.
+        :param description: A description of the experiment.
+        """
+        if not callable(task):
+            raise TypeError("task must be a callable function.")
+        sig = inspect.signature(task)
+        params = sig.parameters
+        if "input_data" not in params:
+            raise TypeError("Task function must have an 'input_data' parameter.")
+        if not isinstance(dataset, Dataset):
```

Comment:
idt we really need to type check each arg. We typically don't as a convention in the rest of the library

Reply:
I thought since if someone doesn't correctly pull/create a Dataset object this will basically break their experiment run anyway this can't hurt to have the assertion here